### PR TITLE
Progres storage v1

### PR DESCRIPTION
**Pull request by innaLebedyk**
Components: 
- common/Link
- common/Input
- form/SignUpForm
- form/SocialLog
- form/SwitcherForm
- pages/signUp

### CODE REVIEW

File: src/components/common/ButtonLink/ButtonLink.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
```
Comment (YuliaSavchuk) [inline src/components/common/ButtonLink/ButtonLink.js]: Add new components to storybook

File: src/utils/window.js [hunk 2]
```diff
@@ -4,3 +4,28 @@ export const getCurrentUrl = () =>
...
+  let result = regex.exec(url);
+
+  if (!result || !result[2]) {
+    return;
+  }
+
+  return decodeURIComponent(result[2].replace(/\+/g, ' '));
+};
+
+export const clareQueryString = () => {
```
Comment (YuliaSavchuk) [inline src/utils/window.js]: typo

File: src/store/questions/actions.js [hunk 3]
```diff
@@ -16,15 +18,21 @@ export const questionsLoaded = questions => dispatch => {
...
+
   dispatch({
     type: QUESTION_ANSWERED,
     payload: { id: question.id, score }
   });
   dispatch(sectionActions.updateProgress(question.sectionId));
   dispatch(courseActions.updateProgress());
+
+  if (saveProgress) {
+    await progressStorage.saveProgress(questionId, response);
```
Comment (YuliaSavchuk) [inline src/store/questions/actions.js]: I think it's better to split actions: updateQuestionProgress(questionId, response); saveProgress()

Comment (oleksandrbulakh) [reply src/store/questions/actions.js]: Done

File: src/utils/window.js (hunk 2 above)
Comment (oleksandrbulakh) [reply src/utils/window.js]: Done

File: src/components/shell/Shell.js [hunk 4]
```diff
@@ -45,10 +62,11 @@ export class Shell extends Component {
       colors,
       fonts,
       isTocExpanded,
-      background
+      background,
+      isUserAuthenticated
     } = this.props;
     const { hasError } = this.state;
-
+    console.log(`is user authenticated: ${isUserAuthenticated}\nBool: ${!isUserAuthenticated}`);
```
[the remark this answers is not in the document]
Comment (oleksandrbulakh) [reply src/components/shell/Shell.js]: Done

**Pull request merged**